pkg/dependency/parser/python/uv/parse.go

```diff
@@ -0,0 +1,164 @@
+		return pkg.Name, pkg
+	})
+}
+
+func (l Lock) directDeps(root Package) map[string]struct{} {
+	deps := make(map[string]struct{})
+	for _, dep := range root.Dependencies {
+		deps[dep.Name] = struct{}{}
+	}
+	return deps
+}
+
+func prodDeps(root Package, packages map[string]Package) map[string]struct{} {
+	visited := make(map[string]struct{})
+	walkPackageDeps(root, packages, visited)
+	return visited
+}
+
+func walkPackageDeps(pkg Package, packages map[string]Package, visited map[string]struct{}) {
+	if _, ok := visited[pkg.Name]; ok {
+		return
+	}
+	visited[pkg.Name] = struct{}{}
+	for _, dep := range pkg.Dependencies {
+		depPkg, exists := packages[dep.Name]
+		if !exists {
+			continue
+		}
+		walkPackageDeps(depPkg, packages, visited)
+	}
+}
+
+func (l Lock) root() (Package, error) {
+	var pkgs []Package
+	for _, pkg := range l.Packages {
+		if pkg.isRoot() {
+			pkgs = append(pkgs, pkg)
+		}
+	}
+
+	// lock file must include root package
+	// cf. https://github.com/astral-sh/uv/blob/f80ddf10b63c3e7b421ca4658e63f97db1e0378c/crates/uv/src/commands/project/lock.rs#L933-L936
+	if len(pkgs) != 1 {
+		return Package{}, xerrors.New("uv lockfile must contain 1 root package")
+	}
+
+	return pkgs[0], nil
+}
+
+type Package struct {
+	Name         string       `toml:"name"`
+	Version      string       `toml:"version"`
+	Source       Source       `toml:"source"`
+	Dependencies []Dependency `toml:"dependencies"`
+}
+
+// https://github.com/astral-sh/uv/blob/f7d647e81d7e1e3be189324b06024ed2057168e6/crates/uv-resolver/src/lock/mod.rs#L572-L579
+func (p Package) isRoot() bool {
+	return p.Source.Editable == "." || p.Source.Virtual == "."
+}
+
+type Source struct {
+	Editable string `toml:"editable"`
+	Virtual  string `toml:"virtual"`
+}
+
+type Dependency struct {
+	Name string `toml:"name"`
+}
+
+type Parser struct{}
+
+func NewParser() *Parser {
+	return &Parser{}
+}
+
+func (p *Parser) Parse(r xio.ReadSeekerAt) ([]ftypes.Package, []ftypes.Dependency, error) {
+	var lock Lock
+	if _, err := toml.NewDecoder(r).Decode(&lock); err != nil {
+		return nil, nil, xerrors.Errorf("failed to decode uv lock file: %w", err)
+	}
+
+	rootPackage, err := lock.root()
+	if err != nil {
+		return nil, nil, err
+	}
+
+	packages := lock.packages()
+	directDeps := lock.directDeps(rootPackage)
+	prodDeps := prodDeps(rootPackage, packages)
+
+	var (
+		pkgs ftypes.Packages
+		deps ftypes.Dependencies
+	)
+
+	for _, pkg := range lock.Packages {
+		if _, ok := prodDeps[pkg.Name]; !ok {
+			continue
+		}
+
+		pkgID := packageID(pkg.Name, pkg.Version)
+		relationship := ftypes.RelationshipIndirect
+		if pkg.isRoot() {
+			relationship = ftypes.RelationshipRoot
+		} else if _, ok := directDeps[pkg.Name]; ok {
+			relationship = ftypes.RelationshipDirect
+		}
+
+		pkgs = append(pkgs, ftypes.Package{
+			ID:           pkgID,
+			Name:         pkg.Name,
+			Version:      pkg.Version,
+			Indirect:     relationship == ftypes.RelationshipIndirect,
```

Comment:
I don't think we need to fill in the deprecated field for the newly added package manager.

Reply:
Done https://github.com/aquasecurity/trivy/pull/8080/commits/602a839e19dbf52fbe30f1ba8638affb73e127f6